huxley/api/mixins.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2011-2015 Berkeley Model United Nations. All rights reserved.
+# Use of this source code is governed by a BSD License (see LICENSE).
+
+import json
+
+from django.http import QueryDict
+
+from rest_framework import status
+from rest_framework.response import Response
+
+class ListUpdateModelMixin(object):
+    """
+    Update a queryset
+    """
+
+    def list_update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', False)
+        # I feel like there should be a cleaner way to do what the next
+        # few lines are trying to do. Any suggestions would be appreciated.
+        data = request.data
+        # This if statement is here for testing purposes
+        if isinstance(data, QueryDict):
+            data = json.loads(request.data.items()[0][0])
+
+        self.lookup_url_kwarg = 'obj_id'
+
+        response_data = []
+        for obj in data:
```

Comment:
We should batch this code, otherwise we're doing two round trips to DB for every single updated object: fetching it, then saving it.

Instead, we should fetch all the objects in one round with the `in` modifier on QuerySet: https://docs.djangoproject.com/en/1.9/ref/models/querysets/#in

Then, we iterate over them and update them inside a transaction, so all the saves occur in one DB request. We do this with the `transaction.atomic` context manager: https://docs.djangoproject.com/en/1.8/topics/db/transactions/#django.db.transaction.atomic An example of this is in the assignment updating code.